api/java/org/openyolo/api/CredentialClientOptions.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 The OpenYOLO Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openyolo.api;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.valid4j.Validation.validate;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import org.openyolo.api.persistence.SettingsFactoryProvider;
+import org.openyolo.api.persistence.internal.SettingsFactoryImpl;
+
+
+/**
+ * Configuration object that may be used to configure a {@link CredentialClient} instance.
+ */
+public class CredentialClientOptions {
+    private final SettingsFactoryProvider mSettingsFactoryProvider;
+    private static final String APP_SETTINGS_NAMESPACE = "app";
+
+    /**
+     * Default and recommended configuration.
+     */
+    public static final CredentialClientOptions DEFAULT =
+            new CredentialClientOptions.Builder().build();
+
+    private CredentialClientOptions(Builder builder) {
+        mSettingsFactoryProvider = builder.mSettingsFactoryProvider;
+    }
+
+    @NonNull
+    AppSettings getAppSettings(Context context) {
+        return AppSettings.getInstance(
+                mSettingsFactoryProvider.get(context, APP_SETTINGS_NAMESPACE));
+    }
+
+    /**
+     * Builder for {@link CredentialClientOptions}.
+     */
+    public static final class Builder {
+        private SettingsFactoryProvider mSettingsFactoryProvider = SettingsFactoryImpl.PROVIDER;
+
+        /**
+         * Specifies the {@link SettingsFactoryProvider} that will be used to ultimately supply the
+         * persistence implementation of settings in {@link AppSettings}.
+         */
+        public Builder setSettingsFactory(

```

Comment:
I think this is overkill - just define a `StateProvider` interface with a `isAutoSignInDisabled` method, and have the developer provide an implementation of that. More methods can be added as needed; we don't need to go as far as defining a full key-to-bool table right now.